client/src/components/TabPane/Cards/MessagesCard.js

```diff
@@ -1,7 +1,462 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { PlusOutlined, EditOutlined, DeleteOutlined } from "@ant-design/icons";
+import { Button, Row, Modal, Input, message, Select  } from "antd";
+import { Table } from "ant-table-extensions";
+import { baseURL } from "../../../config";
 
-const LanguagesCard = () => {
-  return <p>Messages tab TODO</p>;
+const MessagesCard = () => {
+  const [data, setData] = useState();
+  const [messageApi, resultMessageHolder] = message.useMessage();
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [isAdding, setIsAdding] = useState(false);
+  const [editingMessage, setEditingMessage] = useState(null);
+  const [addingMessage, setAddingMessage] = useState(null);
+
+  const [loading, setLoading] = useState(false);
+  const [tableParams, setTableParams] = useState({
+    pagination: {
+      pageSize: 10,
+    },
+  });
+
+  const [langData, setLangData] = useState();
+  const [originalMessData, setoriginalMessData] = useState();
+  const [tagData, setTagData] = useState();
+
+  const success = (message) => {
+    messageApi.open({
+      type: "success",
+      content: `${message}`,
+      style: {
+        marginTop: "150px",
+      },
+      duration: 2,
+    });
+  };
+
+  const error = (message) => {
+    messageApi.open({
+      type: "error",
+      content: `${message}`,
+      style: {
+        marginTop: "150px",
+      },
+      duration: 2,
+    });
+  };
+
+  const onAdd = () => {
+    setIsAdding(true);
+    setAddingMessage({ ...editingMessage });
+  };
+
+  const fetchData = () => {
+    setLoading(true);
+    fetchLanguages();
+    fetchTags();
+    fetch(`${baseURL}/message`)
+      .then((response) => response.json())
+      .then((data) => {
+        for (let i = 0; i < data.length; i++) {
+          const orig = data.find(e => e.id === data[i].original_message);
+          if (orig)
+          {
+            data[i]['original_message_text'] = orig.content;
+          }
+          else
+          {
+            data[i]['original_message_text'] = "";
+          }
+        }
+        setData(data);
+        getOriginalMessages(data);
+        setLoading(false);
+      })
+      .catch((error) => {
+        console.error("Error while fetching data: ", error);
+      });
+  };
+
+  const handleEdit = (record) => {
+    setIsEditing(true);
+    setEditingMessage({ ...record });
+  };
+
+  const handleDelete = (record) => {
+    Modal.confirm({
```

Comment:
I think in case deleting the original message we should add here information that all messages related  to this original message will be deleted

Reply:
Done